### PR TITLE
DOC: roadmap update

### PR DESCRIPTION
**Pull request by rgommers**
- updates the contents for changes in 2020
- reorders sections to reflect priorities from user survey
- makes it easier to understand the concrete goals for "interoperability"

### CODE REVIEW

File: doc/neps/roadmap.rst [hunk 1]
```diff
@@ -17,6 +17,12 @@ facilitate interoperability with all such packages, and the code that uses them,
 may include (among other things) interoperability protocols, better duck typing
 support and ndarray subclass handling.
 
+The key goal is: *make it easy for code written for NumPy to also work with
+other NumPy-like projects.* This will enable GPU support via, e.g, CuPy or JAX,
+distributed array support via Dask, and writing special-purpose arrays (either
+from scratch, or as a ``numpy.ndarray`` subclass) that work well with SciPy,
+scikit-learn and other such packages.
+
```
Comment (mattip) [inline doc/neps/roadmap.rst:25]: This seems like a restatement of the previous paragraph. Do you intend that or would it make sense to merge them?

File: doc/neps/roadmap.rst [hunk 2]
```diff
@@ -33,75 +39,76 @@ of the API, as discussed in `this section of NEP 37
...
-
-
 Performance
 -----------
 
 Improvements to NumPy's performance are important to many users. The primary
 topic at the moment is better use of SIMD instructions, also on platforms other
-than x86 - see :ref:`NEP38`.
+than x86 - see :ref:`NEP38`. The "universal SIMD" infrastructure is in place,
+and we welcome PRs to add SIMD support to NumPy functions.
```
Comment (mattip) [inline doc/neps/roadmap.rst:48]: Improvements to NumPy's performance are important to many users.  We have
focused this effort on Universal SIMD (see :ref:`NEP38`) intrinsics which provide
nice improvements across various hardware platforms via an abstraction layer.
The infrastructure is in place, and we welcome follow-on PRs to add SIMD support
across all relevant NumPy functions.

**Review by mattip — COMMENTED** [review]
The roadmap seems to focus on "ongoing maintainence" more than "new shiny", which is probably the right direction. Maybe worth mentioning a dsire to engage not only as a library but as a team to share knowledge and help shape the general Array API landscape?

File: doc/neps/roadmap.rst (hunk 1 above)
Comment (rgommers) [reply doc/neps/roadmap.rst:25]: I meant to keep it separate. That first paragraph made sense to insiders, but I'm pretty sure it was way too difficult for an average interested user to get from that what the second paragraph now says. So different audiences.

If we can merge them, great - but it's not too easy to do I found.

**Comment by rgommers** [comment]
> The roadmap seems to focus on "ongoing maintainence" more than "new shiny", which is probably the right direction. Maybe worth mentioning a dsire to engage not only as a library but as a team to share knowledge and help shape the general Array API landscape?

That could be useful indeed. Would you make it a separate entry, or include it in the Interoperability section?

Another thing I considered adding, but decided against, was project-related topics like growing the team etc. I figured this roadmap really is about the things we produce, not about the social/community aspects.

**Comment by mattip** [comment]
I think it is a separate entry and something we should try to develop. We could ask for funding to refactor old discussions into white papers, or for contributing to the Data API consortium, or for breadth-wise direct contributions or networking the right people for all the array-API libraries when the have questions about design decisions around edge cases. Maybe the idea is too raw to be in the roadmap right now, but worth thinking about.

**Comment by rgommers** [comment]
> I think it is a separate entry and something we should try to develop. We could ask for funding to refactor old discussions into white papers, or for contributing to the Data API consortium, or for breadth-wise direct contributions or networking the right people for all the array-API libraries when the have questions about design decisions around edge cases. Maybe the idea is too raw to be in the roadmap right now, but worth thinking about.

That all makes sense to me. I think it would need some iteration indeed, so prefer to pick this up in a separate PR.

**Comment by charris** [comment]
Just thought I'd drop [this link](https://ipfs.io/ipfs/QmdA5WkDNALetBn4iFeSepHjdLGJdxPBwZyY47ir1bZGAK/comp/evolution.html) in here for no particular reason except that it tickles my fancy :)

**Comment by charris** [comment]
One thing I got from the Linus thread linked above was the emphasis on feedback and selection.  Getting more and better feedback from users would be useful. Surveys and such help, but are no replacement for participation. @mhvk's feedback had a lot of impact and I wish we had more. Not that I have good ideas of how to do that, but specialized meetings with a wide spectrum of participants seem to help.

**Comment by mattip** [comment]
While this is really only a reformatting and clarification of the already-accepted goals, it feels weird to change the roadmap via a PR without getting more input. Maybe we can use this PR as a starting point for a discussion in a community meeting or on the mailing list?

**Comment by rgommers** [comment]
Happy to have a larger discussion on roadmap in the near future, especially given that I have one left-over deliverable on the CZI grant to update the "scope and vision". Makes sense to me to do that first, then revisit the roadmap.

I don't think it's a hard necessity to do it based on this PR though, this is nothing more than a maintenance update - there's exactly zero important new content, just clarifications, removal of things that were done, and a minor reordering of sections.

**Comment by mattip** [comment]
OK, so if a larger discussion around the roadmap will happen soon, it makes sense to merge this now.

**Comment by mattip** [comment]
Thanks @rgommers

**Comment by mhvk** [comment]
Belatedly, thanks for pointing to the links about linux, @charris. And all the better that the roadmap mentions goals not designs!

**Pull request merged**